### PR TITLE
added intro to pipe/chain/compose (issue #27879)

### PR DESCRIPTION
**Pull request by cormullion**
Added brief intro to this hoping that it will become searchable.

### CODE REVIEW

File: doc/src/manual/functions.md [hunk 1]
```diff
@@ -660,6 +660,61 @@ enclosing scope. For example, the variable `data` in the above example of
...
 can create performance challenges as discussed in [performance tips](@ref man-performance-tips).
 
+## Function composition and piping
+
+Functions in Julia can be combined by composing or piping (chaining) them together.
+
+Function composition is when you combine functions together and apply the resulting composition to arguments.
+You use the function composition operator (`∘`) to compose the functions, so `(f ∘ g)(args...)` is the same as `f(g(args...))`.
+
+You can type the composition operator at the REPL and suitably-configued editors using `\circ<tab>`.
```
Comment (fredrikekre) [inline doc/src/manual/functions.md]: configu**r**ed

File: doc/src/manual/functions.md [hunk 2]
```diff
@@ -660,6 +660,61 @@ enclosing scope. For example, the variable `data` in the above example of
...
+```
+
+Here, the total produced by `sum` is passed to the `sqrt` function. The equivalent composition would be:
+
+```jldoctest
+julia> (sqrt ∘ sum)(1:10)
+7.416198487095663
+```
+
+The elementwise pipe operator `.|>` is a useful combination of the chaining/piping and dot vectorization syntax (described next).
```
Comment (fredrikekre) [inline doc/src/manual/functions.md]: This makes it sound like another operator. How about something like
```
The pipe operator can also be used in combination with broadcasting ....
```

**Comment by cormullion** [comment]
Any more I can do to improve this?

**Review by fredrikekre — APPROVED** [review]

File: doc/src/manual/functions.md [hunk 3]
```diff
@@ -660,6 +660,61 @@ enclosing scope. For example, the variable `data` in the above example of
...
+```
+
+This adds the numbers first, then finds the square root of the result.
+
+The next example composes three functions and maps the result over an array of strings:
+
+```jldoctest
+julia> map(first ∘ reverse ∘ uppercase, split("you can compose functions like this"))
+6-element Array{Char,1}:
+'U'
```
Comment (fredrikekre) [inline doc/src/manual/functions.md]: This fails doctesting, need spaces in front of the elements?

**Pull request merged**